jax/_src/dtypes.py

```diff
@@ -467,8 +467,10 @@ def isdtype(dtype: DTypeLike, kind: str | DType | tuple[str | DType, ...]) -> bo
       options.update(_dtype_kinds[kind])
     elif isinstance(kind, np.dtype):
       options.add(kind)
+    # Check for _ScalarMeta without referencing the class directly
+    elif issubclass(kind.__class__, type) and isinstance(getattr(kind, 'dtype'), np.dtype):
+      options.add(kind.dtype)
```

Comment:
If this is necessary, then the annotations are wrong: `kind` is only allowed to be a `str`, a `dtype` or a tuple thereof. Should `kind` accept `DTypeLike`? If so then the appropriate fix is to call `kind = np.dtype(kind)` to convert it to a `dtype` (so long as it is not one of the documented strings).

Reply:
Yeah good call, I think that makes more sense in general. I've included that change 👍